src-plugins/reformat/shapeBasedInterpolation.cpp

```diff
@@ -460,15 +456,12 @@ shapeBasedInterpolation::~shapeBasedInterpolation()
     d = NULL;
 }
 
-void shapeBasedInterpolation::setInput(MaskType::Pointer input0, MaskType::Pointer input1, MaskType::Pointer input2, MaskType::Pointer input3)
+void shapeBasedInterpolation::setInput(MaskType::Pointer input)
 {
-    if (!input0 || !input1 || !input2 || !input3)
-        return;
-
-    d->input.push_back(input0);
-    d->input.push_back(input1);
-    d->input.push_back(input2);
-    d->input.push_back(input3);
+    if (input)
+    {
+        d->input.push_back(input);
```

Comment:
isn't there an order for setting the inputs? (I saw a distinction between masks and volumes)
If so, I would recommend to use a second argument, like the ```channel``` one we often use.